packages/web-pkg/src/composables/actions/spaces/useSpaceActionsSetIcon.ts

```diff
@@ -0,0 +1,147 @@
+import { SpaceResource } from '@ownclouders/web-client'
+import { computed } from 'vue'
+import { SpaceAction, SpaceActionOptions } from '../types'
+import { useClientService } from '../../clientService'
+import { useLoadingService } from '../../loadingService'
+import { useGettext } from 'vue3-gettext'
+import { useMessages, useModals, useSpacesStore, useUserStore } from '../../piniaStores'
+import { useCreateSpace } from '../../spaces'
+import { buildSpace } from '@ownclouders/web-client/src/helpers'
+import { eventBus } from '../../../services'
+import { Drive } from '@ownclouders/web-client/src/generated'
+import { blobToArrayBuffer, canvasToBlob } from '../../../helpers'
+import EmojiPickerModal from '../../../components/Modals/EmojiPickerModal.vue'
+
+export const useSpaceActionsSetIcon = () => {
+  const userStore = useUserStore()
+  const { showMessage, showErrorMessage } = useMessages()
+  const { $gettext } = useGettext()
+  const clientService = useClientService()
+  const loadingService = useLoadingService()
+  const { createDefaultMetaFolder } = useCreateSpace()
+  const spacesStore = useSpacesStore()
+  const { dispatchModal } = useModals()
+  const handler = ({ resources }: SpaceActionOptions) => {
+    if (resources.length !== 1) {
+      return
+    }
+
+    dispatchModal({
+      elementClass: 'oc-width-auto',
+      title: $gettext('Set icon for %{space}', { space: resources[0].name }),
+      hideConfirmButton: true,
+      customComponent: EmojiPickerModal,
+      focusTrapInitial: 'em-emoji-picker',
+      onConfirm: (emoji: string) => setIconSpace(resources[0], emoji)
+    })
+  }
+
+  const generateEmojiImage = async (emoji: string): Promise<ArrayBuffer | string> => {
+    const canvas = document.createElement('canvas')
+    const context = canvas.getContext('2d')
+    const aspectRatio = 16 / 9,
```

Comment:
It would work just fine for web, but I don't know how other clients implement the image representation, so it might be easier for them to have it in the correct ratio, up to you

Reply:
Let's ask the other clients then. 😁 
@felix-schwarz @TheOneRing @jesmrec would ios/android/desktop work fine with a square space image? i.e. would it be cropped into 16:9 aspect ratio (top and bottom of the emoji cut off) or would it just work?

Reply:
I'd prefer a square image in case of an emoji, but if that would mean work in other clients I'm fine with having "padding" (transparent background) left and right in the image file for a 16:9 aspect ratio of the image.

Reply:
The desktop keeps the aspection ratio and displays it scaled.
In my opinion client should never crop the image...